pkg/sdk/testint/security_integrations_gen_integration_test.go

```diff
@@ -0,0 +1,380 @@
+	jPzQNnQ/zuOu1wvRcVAQmyIIaDQU1V+OVv5vz9G0MNdHUeerRfVuse0i1IlyDtX/
+	sV9lcgU4fIsdwyg0+tyvG8QA8R8mCajy2UDcQS/qh0NB/WGa08tmbedMO5FQ7Obz
+	cBnksmyuq+l4AdbC5nDfK7BSo6CVPQBYLrmsTPKhU+ET50X4IN+nd3NmGlQH8kXo
+	OjU39Udf31fXBDuVC7dfL2uBHAkn9bUV5LwF2bKMeNMRQOrCydgy7jvsO+HrAgMB
+	AAGjUzBRMB0GA1UdDgQWBBT9mt6mehFcEHTTEQcTru4ync3T6DAfBgNVHSMEGDAW
+	gBT9mt6mehFcEHTTEQcTru4ync3T6DAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3
+	DQEBCwUAA4IBAQB177MgJXJAHXbaJ0/KVhWnDDNuZYG+OwzrGaVXiOhXShfxzENc
+	cqsQB4DR7GEIrEicL2xQ23Kg3j7zASmo7T56CZiJ97jIiHDNrhGoAaW+aMhbp6wx
+	WYxLNx9pbaPIORAJ1KEC3hvE4strHJPlQddCYSsXDhIOUTUd71JvR26DHiYQ82TO
+	3wpXHhYdWYZbMjrDDAz0PwdTXyFBuTZxdlTFTxX2lXAE33OsdAFt+oi7JTQh248k
+	0+lmQdhQrSrzhM3WwwuYTEKQVoa2xvWajgqbo7iu2iadWkrxUx/5bjFc5kXej6j7
+	PhfG6C4ddUpAISJhmEViuXq4nVxe0Vk3Efo2
+	`
+
+	createSAML2Integration := func(t *testing.T, siID sdk.AccountObjectIdentifier, with func(*sdk.CreateSAML2SecurityIntegrationRequest)) {
+		t.Helper()
+		_, err := client.ExecForTests(ctx, "ALTER ACCOUNT SET ENABLE_IDENTIFIER_FIRST_LOGIN = true")
+		require.NoError(t, err)
+
+		saml2Req := sdk.NewCreateSAML2SecurityIntegrationRequest(siID, false, "test", "https://example.com", "Custom", x509)
+		if with != nil {
+			with(saml2Req)
+		}
+		err = client.SecurityIntegrations.CreateSAML2(ctx, saml2Req)
+		require.NoError(t, err)
+		cleanupSecurityIntegration(t, siID)
+	}
+
+	createSCIMIntegration := func(t *testing.T, siID sdk.AccountObjectIdentifier, with func(*sdk.CreateSCIMSecurityIntegrationRequest)) {
+		t.Helper()
+		roleID := sdk.NewAccountObjectIdentifier("GENERIC_SCIM_PROVISIONER")
+		err := client.Roles.Create(ctx, sdk.NewCreateRoleRequest(roleID).WithIfNotExists(true))
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			err = client.Roles.Drop(ctx, sdk.NewDropRoleRequest(roleID))
+			assert.NoError(t, err)
+		})
+		currentRole := testClientHelper().Context.CurrentRole(t)
+		err = client.Roles.Grant(ctx, sdk.NewGrantRoleRequest(roleID, sdk.GrantRole{Role: sdk.Pointer(sdk.NewAccountObjectIdentifier(currentRole))}))
+		require.NoError(t, err)
+
+		scimReq := sdk.NewCreateSCIMSecurityIntegrationRequest(siID, false, &sdk.SCIMSecurityIntegrationSCIMClientGeneric, &sdk.SCIMSecurityIntegrationRunAsRoleGenericScimProvisioner)
+		if with != nil {
+			with(scimReq)
+		}
+		err = client.SecurityIntegrations.CreateSCIM(ctx, scimReq)
+		require.NoError(t, err)
+		cleanupSecurityIntegration(t, siID)
+	}
+
+	assertSecurityIntegration := func(t *testing.T, si *sdk.SecurityIntegration, id sdk.AccountObjectIdentifier, siType string, enabled bool, comment string) {
+		t.Helper()
+		assert.Equal(t, id.Name(), si.Name)
+		assert.Equal(t, siType, si.IntegrationType)
+		assert.Equal(t, enabled, si.Enabled)
+		assert.Equal(t, comment, si.Comment)
+		assert.Equal(t, "SECURITY", si.Category)
+	}
+
+	assertSCIMDescribe := func(details []sdk.SecurityIntegrationProperty, enabled, networkPolicy, runAsRole, syncPassword, comment string) {
```

Comment:
only so many?

Reply:
I double checked and here I assert all properties returned by describe operation. Note that there's a separate generic check for show operation. I think that this func can call `assertSecurityIntegration` so tests call assertions only once.